### PR TITLE
Added: Documentation for TreeDataGrid Usage, Including Examples

### PR DESCRIPTION
**Pull request by Sewer56**
Induction

Including examples you can compile and inspect.

### CODE REVIEW

**Review by erri120 — CHANGES_REQUESTED** [review]
The code in the documentation is duplicated in the example. The code should only be in the example. To prevent the documentation from becoming stale and outdated, code should only be in the example.

The documentation should explain the concept and then refer to the example.

**Comment by Al12rs** [comment]
Just a heads up, I'd like to have a look as well before we merge

**Comment by Al12rs** [comment]
Had my quick readthrough, just the comment above from me.

File: docs/development-guidelines/UICodingGuidelines.md [hunk 1]
```diff
@@ -1475,6 +1475,409 @@ this.WhenActivated(disposables =>
...
 
+### Trees with Columns (TreeDataGrid)
+
+!!! info "Functional Example(s) available at [TreeDataGrid Examples][treedatagrid examples]"
+
+#### Basic TreeDataGrid
+
+!!! info "Functional Example(s) available at [TreeDataGrid Examples][treedatagrid basic] as 'Basic'"
+
+To create a TreeDataGrid, we will take the following steps:
```
[the remark this answers is not in the document]
Comment (Sewer56) [reply docs/development-guidelines/UICodingGuidelines.md:1486]: Actually a very good point, I'll get that sorted.

**Review by erri120 — APPROVED** [review]

**Pull request merged**